bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.net.http;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.Socket;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PEMTrustManager} is a {@link X509ExtendedTrustManager} implementation which loads a certificate in
+ * PEM format and validates it against the servers certificate.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class PEMTrustManager extends X509ExtendedTrustManager {
+
+    public static final String BEGIN_CERT = "-----BEGIN CERTIFICATE-----" + System.getProperty("line.separator");
+    public static final String END_CERT = System.getProperty("line.separator") + "-----END CERTIFICATE-----"
+            + System.getProperty("line.separator");
+
+    private final X509Certificate trustedCert;
+
+    /**
+     * Creates a {@link PEMTrustManager} instance by passing the PEM certificate as {@link String}.
+     * The PEM format typically starts with <code>"-----BEGIN CERTIFICATE-----"</code> and ends with
+     * <code>"-----END CERTIFICATE-----"</code>. The base 64 encoded certificate information are placed in between.
+     *
+     * @param pemCert the PEM certificate
+     * @throws CertificateException
+     */
+    public PEMTrustManager(String pemCert) throws CertificateException {
+        if (!pemCert.isBlank() && pemCert.startsWith(BEGIN_CERT)) {
```

Comment:
One of the use-cases (e.g. with the http binding) could be that there is a configuration field in the thing config that allows pinning the certificate permanently (the `getInstanceFromServer` does only pin until restarting the system). 

Using `line.separator` would require the configuration  to be different depending on the operating system that openHAB runs on and could lead to problems if the user configures on a different OS. Maybe the cert string should be normalized to the `line.separator` line endings before the checks are done.

Reply:
Can you check if the new version handles the line endings even better?